File: Python/Algorithms/942.py
class Solution:
    def diStringMatch(self, S: str) -> List[int]:
        res = []
        append = res.append
        i, j = 0, len(S)
        for c in S:
            if c == 'I':
                append(i)
                i += 1
            else:
                append(j)
                j -= 1
        append(j)
        return res

File: Rust/Algorithms/942.rs
impl Solution {
    pub fn di_string_match(s: String) -> Vec<i32> {
        let mut res = Vec::with_capacity(s.len() + 1);
        let (mut i, mut j) = (0, s.len() as i32);
        for c in s.chars() {
            match c {
                'I' => {
                    res.push(i);
                    i += 1;
                }
                'D' => {
                    res.push(j);
                    j -= 1;
                }
                _ => (),
            }
        }
        res.push(j);
        res
    }
}
